test/worker_test.rb

```diff
@@ -654,10 +661,12 @@ def self.perform
     workerA.register_worker
 
     assert_equal 1, Resque.workers.size
+    assert_equal({}, Resque::Worker.all_heartbeats)
 
     @worker.prune_dead_workers
 
     assert_equal 1, Resque.workers.size
+    assert_equal({}, Resque::Worker.all_heartbeats)
```

Comment:
Why bother with this assertion?  There wasn't any heartbeat to remove in this case.  Are you worried that prune_dead_workers will set a heartbeat?